core/src/main/scala/com/codacy/analysis/core/storage/FileDataStorage.scala

```diff
@@ -0,0 +1,55 @@
+package com.codacy.analysis.core.storage
+
+import better.files.File
+import io.circe._
+import io.circe.syntax._
+import org.log4s.{Logger, getLogger}
+
+import scala.util.Try
+
+trait FileDataStorage[T] {
+  private val logger: Logger = getLogger
+
+  implicit val encoder: Encoder[T]
+  implicit val decoder: Decoder[T]
+
+  def storageFilename: String
+
+  val storageFile: File = File.currentWorkingDirectory / ".codacy" / "codacy-analysis-cli" / storageFilename
+
+  private def writeToFile(file: File, content: String): Try[File] =
+    synchronized {
+      Try {
+        file.createFileIfNotExists(createParents = true)
+        file.write(content)
+      }
+    }
+
+  private def readFromFile(file: File): String =
+    synchronized {
+      file.contentAsString
+    }
+
+  def invalidate(): Boolean =
+    synchronized {
+      logger.debug("Invalidating storage")
+      !storageFile.delete().exists
```

Comment:
If the file doesn't exist this will throw an exception. We should wrap this on a Try instead of the exists. Or check if it exists before deleting it.

Reply:
Wrapped this on a Try 👍